### PR TITLE
Upgrade to OpenCV==4.10.0

### LINKED ISSUE #66: test not good in my dataset about fish images
reproject error:
image

other message:
image

I don't know if it's normal.

my dataset and configs link:
https://drive.google.com/file/d/1OOtaLzSQAoGMMsNgfCF0Aaujv5h6Ac0H/view?usp=drive_link

### PR DESCRIPTION
**Pull request by BAILOOL**
Closes #66

### CODE REVIEW

**Review by rameau-fr — DISMISSED** [review]
Great! Is the DockerHub updated too?

File: apps/create_charuco_boards/src/create_charuco_boards.cpp [hunk 1]
```diff
@@ -59,27 +59,28 @@ int main(int argc, char *argv[]) {
...
-      int id_offset = charucoBoards[i - 1]->ids.size() + offset_count;
+      int id_offset = charucoBoards[i - 1].getIds().size() + offset_count;
       offset_count = id_offset;
-      for (auto &id : charuco->ids) {
-        id += id_offset;
-      }
+      // for (auto &id : charuco->getIds()) {
+      //   id += id_offset;
+      // }
     }
```
Comment (BAILOOL) [inline apps/create_charuco_boards/src/create_charuco_boards.cpp:78]: do you remember why was this needed at that time, @rameau-fr? It looks like some manipulation with indices of the first board. However, In the current version, the index vector is const so cannot be modified.

File: McCalib/src/McCalib.cpp [hunk 2]
```diff
@@ -117,15 +117,16 @@ Calibration::Calibration(const std::string config_path) {
...
     if (i != 0) // If it is the first board then just use the standard idx
     {
-      int id_offset = charuco_boards[i - 1]->ids.size() + offset_count;
+      int id_offset = charuco_boards[i - 1]->getIds().size() + offset_count;
       offset_count = id_offset;
-      for (auto &id : charuco->ids)
-        id += id_offset;
+      // for (int &id : charuco.getIds())
+      //   id += id_offset;
     }
```
Comment (BAILOOL) [inline McCalib/src/McCalib.cpp:130]: Same comment as https://github.com/rameau-fr/MC-Calib/pull/80#discussion_r1739301792 @rameau-fr

File: apps/create_charuco_boards/src/create_charuco_boards.cpp (hunk 1 above)
Comment (BobbleLaw) [reply apps/create_charuco_boards/src/create_charuco_boards.cpp:78]: This part also confused me the first time I read it. I believe the intention is to support multiple boards. In interpolateCornersCharuco (or CharucoDetector::detectBoard in OpenCV 4.10.0), it checks if the detected markerIds are present in the CharucoBoard's IDs. In OpenCV 4.10.0, the implementation of CharucoBoard is hidden using the Pimpl idiom, but we can still specify the IDs in its constructor.

Comment (BAILOOL) [reply apps/create_charuco_boards/src/create_charuco_boards.cpp]: @BobbleLaw, thank you for the explanation. Fixed in a33577ce13e7165e54536689600b0f38d8bbd8ac

File: McCalib/src/McCalib.cpp (hunk 2 above)
Comment (BAILOOL) [reply McCalib/src/McCalib.cpp:142]: Fixed in a33577ce13e7165e54536689600b0f38d8bbd8ac

File: apps/create_charuco_boards/src/create_charuco_boards.cpp (hunk 1 above)
Comment (BobbleLaw) [reply apps/create_charuco_boards/src/create_charuco_boards.cpp]: It's okay. I refactored MC-Calib a lot to use in my project, so gonna keep my eyes on the changes from original authors (which are you guys), in case I misunderstand anything :rofl:.

**Comment by BAILOOL** [comment]
@rameau-fr with OpenCV=4.10.0 we have some unit tests failing. It's probably related to [hand-eye calibration not running](https://github.com/rameau-fr/MC-Calib/actions/runs/10677026306/job/29591569976?pr=80#step:6:3361).

The change to OpenCV was introduced in https://github.com/opencv/opencv/pull/24897 where a part of the change was hard-coded filtering (with no possibility to change the thresholds).

```bash
if (Pgij_norm < 0.3 || Pcij_norm < 0.3 || Pgij_norm > 1.7 || Pcij_norm > 1.7) {
    continue;
}
```

**Comment by shreyas23** [comment]
Are there any expected quality upgrades from this PR?

**Comment by BAILOOL** [comment]
@shreyas23 TSAI hand-eye calibration method seems to be improved in the latest OpenCV. However, after the upgrade, our unit tests, where the accuracy is checked against ground truth synthetic sequences, are failing.

Potentially, there is a bug in our code (#77, #56) and we will try to investigate it further when we have time.

Meanwhile, if you know of bugs or improvements to the codebase, feel free to list them. PR contributions are also welcomed.

**Comment by BAILOOL** [comment]
Seems that backwards compatibility can be achieved by setting `board.setLegacyPattern(true);` as mentioned in the issue https://github.com/opencv/opencv/issues/23152

**Comment by BAILOOL** [comment]
Unit tests still do not pass. Potentially, there is more divergence between OpenCV versions despite setting `board.setLegacyPattern(true)` https://github.com/opencv/opencv/issues/26126
